packages/clerk-js/src/core/clerk.ts

```diff
@@ -1137,16 +1137,23 @@ export default class Clerk implements ClerkInterface {
       clerkMissingSignInUrlAsSatellite();
     }
 
-    return this.#instanceType === 'development';
+    // Attempt to sync if user is logged out or session has expired
+    const cookieHandler = createCookieHandler();
+    return (
+      this.#instanceType === 'development' &&
+      (!this.#devBrowserHandler?.getDevBrowserJWT() ||
+        cookieHandler.getClientUatCookie() <= 0 ||
+        (this.#authService?.isSessionExpired ?? true))
```

Comment:
🔧 Consider moving the nulling coalesce operator inside the isSessionExpired to simplify the logic and the readability of this conditional expression.

Reply:
Unfortunately this will not be much of help in this case.

this.#authService.isSessionExpired is a getter that always return a boolean. The only case that `isSessionExpired` might be undefined is when `this.#authService` is undefined. This shall never happen tho. The `?? true` is a safeguard I added so that if for whatever reason we cannot determine if session is expired we should sync